lib/accessories/thermostat.js

```diff
@@ -115,31 +127,20 @@ FritzThermostatAccessory.prototype.setTargetHeatingCoolingState = function(state
         return;
     }
 
+    var service = this.services.Thermostat;
     this.platform.log(`Setting ${this.type} ${this.ain} heating state`);
-    var self = this, promise;
-
-    switch (state) {
-        case Characteristic.TargetHeatingCoolingState.OFF:
-        case Characteristic.TargetHeatingCoolingState.COOL:
-            promise = this.platform.fritz('getTempNight', this.ain);
-            break;
-        case Characteristic.TargetHeatingCoolingState.HEAT:
-            promise = this.platform.fritz('getTempComfort', this.ain);
-            break;
-        case Characteristic.TargetHeatingCoolingState.AUTO:
-            // let the Fritz!Box do its job
-            return;
-    }
 
-    promise.then(function(temp) {
-        if (temp === undefined) {
-            return;
-        }
 
-        self.platform.fritz('setTempTarget', self.ain, temp).then(function(temp) {
-            service.getCharacteristic(Characteristic.TargetTemperature).setValue(temp, undefined, FritzPlatform.Context);
-        });
-    });
+    let targetTemperatureString = 'off';
+    if (state === Characteristic.TargetHeatingCoolingState.OFF) {
+        targetTemperatureString = 'off';
+    } else if (state === Characteristic.TargetHeatingCoolingState.HEAT) {
+        targetTemperatureString = this.services.Thermostat.fritzTargetTemperature;
+    }
+
+    service.getCharacteristic(Characteristic.TargetHeatingCoolingState).setValue(state, undefined, FritzPlatform.Context);
+    service.getCharacteristic(Characteristic.CurrentHeatingCoolingState).updateValue(this.calculateCurrentHeatingCoolingState());
```

Comment:
Spannend- updateValue kannte ich noch nicht. Warum nicht setValue?

Reply:
Using setValue will trigger the ‘set’ event in your plugin, so use it where it makes sense.  If you don’t have a ‘set’ event in your plugin it works the same as updateValue.

Reply:
Wie bereits erläutert löst updateValue kein `set` event aus (sondern nur ein `change` event, sofern sich der Wert ändern; kenne kein Plugin das auf dieses Event subscribed lol). Da CurrentHeatingCoolingState sowieso ein read-only event ist haben wir sowieso keinen set listener. Also wäre es sogar egal. Jedoch ist es so im Code klarer zu verstehen was wir hier bezwecken wollen (und zwar kein potentielles set event zu callen).
Im Plugin wird meines wissen immer nur auf setValue zurückgegriffen, was meine Meinung den Code wirklich schwer leserlich macht und zu Frustrationen führt. Vor allem an stellen wie der update function werden dadurch Sachen doppelt und dreifach gecalled. Ich wollte mit diesem PR jedoch kein Code Cleanup veranstalten (da es darum ja gar nicht geht) und habe mich an den meisten Stellen dem aktuell etablierten Code style angepasst.